keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
-## ProvisionerGetInfo
+The API of `bucket.json` follows the same versioning and lifecycle as the rest of Bucket* APIs. Workloads are expected to read the definition in this file to access a bucket.
 
-This call is meant to retrieve the unique provisioner Identity. This identity will have to be set in `BucketRequest.Provisioner` field in order to invoke this specific provisioner.
+# COSI API Reference
 
-```
-message ProvisionerGetInfoRequest {
-    // Intentionally left blank
-}
+## Bucket 
 
-message ProvisionerGetInfoResponse {
-  // The name MUST follow domain name notation format
-  // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD
-  // include the plugin's host company name and the plugin name,
-  // to minimize the possibility of collisions. It MUST be 63
-  // characters or less, beginning and ending with an alphanumeric
-  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
-  // alphanumerics between. This field is REQUIRED.
-  string name = 1;
+Resource to represent a Bucket in OSP. Buckets are cluster scoped.
+
+```yaml
+Bucket {
+  TypeMeta
+  ObjectMeta
+  
+  Spec BucketSpec {
+    // Provisioner is the name of driver associated with this bucket
+    Provisioner string 
+
+    // DeletionPolicy is used to specify how COSI should handle deletion of this 
+    // bucket. There are 3 possible values:
+    //  - Retain: Indicates that the bucket should not be deleted from the OSP
+    //  - Delete: Indicates that the bucket should be deleted from the OSP
+    //        once all the workloads accessing this bucket are done
+    //  - ForceDelete: Indicates that the bucket and its contents should be 
+    //        deleted from the OSP even if workloads are currently accessing 
+    //        this bucket
+    DeletionPolicy DeletionPolicy 
+    
+    // Name of the BucketClass specified in the BucketRequest
+    // +optional
+    BucketClassName string 
+    
+    // Name of the BucketRequest that resulted in the creation of this Bucket
+    // +optional
+    BucketRequest corev1.ObjectReference
+    
+    // AllowedNamespaces is the list of namespaces from which workloads  
+    // can access this Bucket. If empty, all namespaces can access this access
+    // +optional 
+    AllowedNamespaces []string 
+
+    // Protocol is the data API this bucket is expected to adhere to.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocol Protocol
+    
+    // Parameters is an opaque map for passing in configuration to a driver
+    // for creating the bucket
+    // +optional
+    Parameters map[string]string 
+  }    
+  
+  Status BucketStatus {
+    // BucketAvailable indicates that the bucket is ready for consumption
+    // by workloads
+    BucketAvailable bool
+    
+    // BucketID is the unique id for the bucket in the OSP
+    BucketID string
+  }
```

Comment:
In the brownfield use case where the admin creates a `Bucket` instance, there doesn't seem to be a way to point to the backend bucket. In the earlier KEP `spec.bucketID` supported this but I don't see how it works in this PR.
Perhaps this use case is not supported?